src/lib.rs

```diff
@@ -394,26 +401,22 @@ impl Message {
     /// let signer: Vec<u8> = message.verify_eip191(&signature)?;
     /// ```
     pub fn verify_eip191(&self, sig: &[u8; 65]) -> Result<Vec<u8>, VerificationError> {
-        use k256::{
-            ecdsa::{
-                recoverable::{Id, Signature},
-                signature::Signature as S,
-                Signature as Sig,
-            },
-            elliptic_curve::sec1::ToEncodedPoint,
-        };
-        use sha3::{Digest, Keccak256};
-        let pk = Signature::new(&Sig::from_bytes(&sig[..64])?, Id::new(&sig[64] % 27)?)?
-            .recover_verifying_key(&self.eip191_bytes()?)?;
+        let msg = self.eip191_bytes()?;
+        let signature: Signature = Signature::from_slice(&sig[..64])?;
+        let recovery_id = RecoveryId::try_from(&sig[64] % 27)?;
 
-        if Keccak256::default()
+        let pk: VerifyingKey = VerifyingKey::recover_from_msg(&msg, &signature, recovery_id)?;
```

Comment:
this line implicitly uses sha2 instead of keccak256. you need to hash the message with keccak256 and then use `recover_from_digest`

Reply:
or `recover_from_prehash`

Reply:
citations:

https://docs.rs/ecdsa/0.16.6/ecdsa/struct.VerifyingKey.html#method.recover_from_msg
https://docs.rs/k256/latest/src/k256/ecdsa.rs.html#182-184
https://docs.rs/ecdsa/0.16.6/src/ecdsa/recovery.rs.html#281-316

Reply:
recover_from_prehash did it! thank you!